seawatch_registration/views/profile.py

```diff
@@ -39,31 +41,47 @@ def form_valid(self, form):
         return super().form_valid(form)
 
 
-class DetailView(LoginRequiredMixin, UserPassesTestMixin, generic.DetailView):
+class DetailView(LoginRequiredMixin, HasProfileMixin, generic.DetailView):
     model = Profile
     nav_item = 'profile'
 
     def get_object(self):
         return Profile.objects.get(user=self.request.user)
 
-    def test_func(self):
-        return Profile.objects.filter(user=self.request.user).exists()
 
-
-class UpdateView(LoginRequiredMixin, UserPassesTestMixin, RedirectNextMixin, generic.UpdateView):
+class UpdateView(LoginRequiredMixin, RedirectNextMixin, generic.View):
```

Comment:
Better use TemplateView and `get_context_data()`.

Reply:
Passt das jetzt so?

Reply:
Ja.